projects/vdk-plugins/vdk-structlog/README.md

```diff
@@ -0,0 +1,117 @@
+# Structured Logging For VDK
+
+This plugin allows users to:
+- select the log output format
+- configure the logging metadata
+- display metadata added by bound loggers
+
+## Usage
+
+```
+pip install vdk-structlog
+```
+
+### Configuration
+
+(`vdk config-help` is useful command to browse all config options of your installation of vdk)
+
+| Name             | Description                                                           | (example)  Value                                                                                                   |
+| ---------------- | --------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------ |
+| logging_metadata | Configure the metadata that will be output along with the log message | "timestamp, level, logger_name, file_name, line_number, function_name, vdk_job_name, vdk_step_name, vdk_step_type" |
+| logging_format   | Configure the logging output format. Available formats: json, console | "console"                                                                                                          |
+
+### Example: Configure metadata
+
+Create a data job and add the following config options
+
+```
+[vdk]
+logging_metadata=timestamp,level,file_name,vdk_job_name
+logging_format=console
+```
+
+Then run the data job. You should see just the configured tags where relevant.
+For example, you won't see the vdk_job_name outside of log statements directly
+related to job execution.
+
+```
```

Comment:
Use collapsable blocks for the logs to make easier to read
 
  In Markdown this happens with <details> and <summary> tags. For example:
```
<details>
 <summary>Click to expand</summary>
 <p>This is the content inside the collapsible block. You can put text, images, or any other HTML elements here.</p>
</details>
```

Reply:
Couldn't get this to work.

![Screenshot 2023-10-20 at 18 27 02](https://github.com/vmware/versatile-data-kit/assets/91800778/6b302e43-e8e2-426b-8b5b-ae2d2570bf3c)

The snippets themselves aren't that long, so I won't spend more time on it.